From 9ddc884069181894a5d0a5da790610cee6167eae Mon Sep 17 00:00:00 2001
From: Vincent Prince <vincent.prince.fr@gmail.com>
Date: Thu, 30 May 2024 02:21:02 -0700
Subject: [PATCH] mongodb: skip until python 3.12 fixes are available.

Hello

It seems to compile with this patch

Upstream-Status: Rejected [obsolete version]

Patch from: https://lists.openembedded.org/g/openembedded-devel/message/110624

Signed-off-by: Mark Hatle <mark.hatle@amd.com>
---
 buildscripts/moduleconfig.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/buildscripts/moduleconfig.py b/buildscripts/moduleconfig.py
index b4d0bba0490..f59ddd7bc5c 100644
--- a/buildscripts/moduleconfig.py
+++ b/buildscripts/moduleconfig.py
@@ -27,7 +27,7 @@ MongoDB SConscript files do.
 __all__ = ('discover_modules', 'discover_module_directories', 'configure_modules',
            'register_module_test')  # pylint: disable=undefined-all-variable
 
-import imp
+import importlib
 import inspect
 import os
 
@@ -71,7 +71,7 @@ def discover_modules(module_root, allowed_modules):
             print("adding module: %s" % (name))
             fp = open(build_py, "r")
             try:
-                module = imp.load_module("module_" + name, fp, build_py,
+                module = importlib.load_module("module_" + name, fp, build_py,
                                          (".py", "r", imp.PY_SOURCE))
                 if getattr(module, "name", None) is None:
                     module.name = name
-- 
2.34.1

